app/util/networks/index.js

```diff
@@ -501,3 +507,46 @@ export const getBlockExplorerTxUrl = (
  */
 export const getIsNetworkOnboarded = (chainId, networkOnboardedState) =>
   networkOnboardedState[chainId];
+
+/**
+ * Convert the given value into a valid network ID. The ID is accepted
+ * as either a number, a decimal string, or a 0x-prefixed hex string.
+ *
+ * @param value - The network ID to convert, in an unknown format.
+ * @returns A valid network ID (as a decimal string)
+ * @throws If the given value cannot be safely parsed.
+ */
+export function convertNetworkId(value) {
+  if (typeof value === 'number' && !Number.isNaN(value)) {
+    return `${value}`;
+  } else if (isStrictHexString(value)) {
+    return `${convertHexToDecimal(value)}`;
+  } else if (typeof value === 'string' && /^\d+$/u.test(value)) {
+    return value;
+  }
+  throw new Error(`Cannot parse as a valid network ID: '${value}'`);
+}
+/**
+ * This function is only needed to get the networkId to support some legacy networks
```

Comment:
Nit: Here is my attempt to clarify the purpose of this; it's not for legacy networks, but for dapps that are using deprecated provider properties/events.

```suggestion
 * This function is only needed to get the `networkId` to support the deprecated `networkVersion` provider property and the deprecated `networkChanged` provider event.
 *
```